lib/src/models/timer.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:timezone/data/latest.dart' as tz;
+import 'package:timezone/timezone.dart' as tz;
+
+import '../util/notification_service.dart';
+
+class TimerList {
+  static final TimerList _timerList =
+  TimerList._internal();
+  List<Timer> timers = <Timer>[];
+
+  factory TimerList() {
+    return _timerList;
+  }
+
+  TimerList._internal();
+
+  clear() {
+    this.timers.clear();
+    NotificationService().cancelAll();
+  }
+}
+
+class Timer {
+  final String title;
+  final String body;
+  final Duration duration;
+  int id;
+  tz.TZDateTime done;
+  final int recipeId;
+
+  Timer(this.recipeId, this.title, this.body, this.duration, [tz.TZDateTime done]) {
+    tz.initializeTimeZones();
+    this.done = tz.TZDateTime.now(tz.local).add(this.duration);
+    TimerList().timers.add(this);
+  }
+
+  // Restore Timer fom pending notification
+  Timer.restore(this.recipeId, this.title, this.body, this.duration, this.done, this.id) {
+    TimerList().timers.add(this);
+  }
+  factory Timer.fromJson(Map<String, dynamic> json, int id) {
+    tz.initializeTimeZones();
+    return Timer.restore(
+        json['recipeId'],
+        json['title'],
+        json['body'],
+        new Duration(minutes: json['duration']),
+        tz.TZDateTime.fromMicrosecondsSinceEpoch(tz.local, json['done']),
+        id
+    );
+  }
+
+  Map<String, dynamic> toJson() => {
+    'title': title,
+    'body': body,
+    'duration': duration.inMinutes,
+    'done': done.microsecondsSinceEpoch,
+    'id': id,
+    'recipeId': recipeId,
+  };
+
+  show() async {
+    NotificationService().show(this);
+  }
+
+  // cancel the timer
+  cancel() {
+    NotificationService().cancel(this);
+    TimerList().timers.remove(this);
+  }
+
+  // calculate progress
+  double completed() {
```

Comment:
```suggestion
  double progress() {
```

Reply:
completed is a verb so it implicates that we do an action. I would use a noun as you want to get the state of the object not manipulate the object.

Reply:
completedTime would also work